### ISSUE TITLE
it would be handy if $spack worked in environment's spack.yaml file

### ISSUE THREAD

**Issue by hartzell** [LLNL/spack#16264]
It would be handy if $spack worked in environment's spack.yaml file, as it does in `config.yaml`.

### Rationale

I'd like to be able to refer to the root of the spack installation when describing the root of a view in an environment's `spack.yaml`.

Current choices seem to absolute paths and relative paths (which seem to be relative the the cwd when `spack install` is run.

When I tried using `$spack/foo` I ended up with a view rooted at the literal `$spack/foo`.

### General information

- [x] I have run `spack --version` and reported the version of Spack
   - `0.14.1-684-f912cce7e`
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by haampie** [comment]
Also relative paths to repo's would be useful, since `develop` now also supports relative paths to the spack.yaml file

```
spack:
  specs:
  - mypkg@main
  view: false
  develop:
    mypkg:
      path: ./mypkg
      spec: mypkg@main
  repos:
  - ./mypkg/spack
```

Especially useful when switching between docker and host where paths may be different, but the relative path is the same.

**Comment by becker33** [comment]
@haampie, where in the spack.yaml file did `$spack` not work that was causing a problem? All of the config values that could come from a config file should be read the same way from spack.yaml as from *.yaml config files.

**Comment by becker33** [comment]
Fixed by #21996

**Comment by haampie** [comment]
Thanks @becker33, very useful!
